### PR TITLE
Hypothesis Testing in Data Science

### PR DESCRIPTION
**Pull request by collins-a**
Article as index.md, hero image as hero.jpg, other images include nullalternative.png and significancelevel.png

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [ ] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by collins-a** [comment]
This fixes #1023

File: articles/hypothesis/index.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+Even though most of the concepts we cover in this article are predominantly statistical, it is important to understand how the term hypothesis is perceived from either a purely statistical or machine learning perspective.
+
+When carrying out statistical hypothesis tests, we attempt to calculate the critical value, which shall be covered later on. We can refer to this critical value as an effect.
+The interpretation of the critical value is significant. It determines the likelihood of observing the effect if observations do not have a relationship. The suggestion that the effect is real comes about if the aforementioned likelihood is minute. If the likelihood is large, the effect is likely not to be real. In statistical hypothesis tests, there is no comment on the size of the effect. These tests are concerned with how likely the effect is present or absent in the population in consideration. This is based on the observed data samples.
+Statistical hypotheses are thus based on identifying the relationships between observations. They are probabilistic explanations about these relationships. Null and alternative hypotheses are denoted as H0 and Ha respectively.
```
Comment (lalith1403) [inline articles/hypothesis/index.md]: $H_0 and H_a$. I will make this change for now. However, note this change.

File: articles/hypothesis/index.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+**Type II error**. We can define a type II error as a situation where we retain a null hypothesis, but it is false. It is denoted by beta.
+
+#### Find the region of rejection for the null hypothesis
+
+There exists a region in the sample space where we reject the null hypothesis. The rejection is if a calculated value lies in the region. This region is known as the critical region. We may reference the image in the previous section. A normal curve depicts the critical region in hypothesis testing. In the case of a type I error, it is referred to as the alpha region. It is a beta region in the context of a type II error.
+
+#### Compute p-value
+
+Assuming the null hypothesis is true, the probability of getting an outcome at least as extreme as the observed outcome of a hypothesis test is the p-value. The meaning of “extreme” is dependent on how the testing is carried out. The p-value determines whether there is enough evidence to retain the alternative hypothesis or retain the null hypothesis.
```
Comment (lalith1403) [inline articles/hypothesis/index.md:75]: There are several tests to compute the critical region. Why did you go for the p-test? z-test, f-test, and chi-squared test are other possible tests as well. In this example, the p-test makes sense as we are comparing two distributions. However, given the mean or standard deviation of the distributions, scenarios change. I think if we can just brief about that, the reader will understand the significance of these tests,

File: articles/hypothesis/index.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+### Steps to Test a Hypothesis
+
+A hypothesis test evaluates two statements about a population. The statements are mutually exclusive. The test concludes which statement best reflects the sample data. A hypothesis test helps us determine the statistical significance of a finding. We say a finding is statistically significant when its likelihood of occurrence is very low, given the null hypothesis. This section describes the steps to test a hypothesis as we define the concepts involved in the testing process.
+
+#### Establish hypotheses
+
+The first step in the testing of a hypothesis is first defining a hypothesis. This is done by establishing both a null and alternative hypothesis. A null hypothesis can be thought of as a statement claiming no relationship between two measured events. It is an assumption made, which may be based on domain experience. Scientists carry out experiments to retain or reject a null hypothesis based upon the nature of (or lack of) the relationship between occurrences. A null hypothesis is usually considered to be true until proven otherwise. It is denoted as H0.
```
Comment (lalith1403) [inline articles/hypothesis/index.md:43]: $H_0$

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Amazing article. Well done.

File: articles/hypothesis/index.md (hunk 1 above)
Comment (collins-a) [reply articles/hypothesis/index.md]: Thank you for this. I'll be keener

**Comment by collins-a** [comment]
@lalith1403 thank you for the review. I've made some changes. Cheeers!

**Review by lalith1403 — APPROVED** [review]
Well written article.

**Comment by hectorkambow** [comment]
plagiarism check done

**Pull request merged**